kotlinx-coroutines-core/common/src/channels/ConflatedChannel.kt

```diff
@@ -15,80 +13,118 @@ import kotlinx.coroutines.internal.*
  * Sender to this channel never suspends and [offer] always returns `true`.
  *
  * This channel is created by `Channel(Channel.CONFLATED)` factory function invocation.
- *
- * This implementation is fully lock-free.
  */
 internal open class ConflatedChannel<E> : AbstractChannel<E>() {
-    protected final override val isBufferAlwaysEmpty: Boolean get() = true
-    protected final override val isBufferEmpty: Boolean get() = true
-    protected final override val isBufferAlwaysFull: Boolean get() = false
+    protected final override val isBufferAlwaysEmpty: Boolean = false
+    protected final override val isBufferEmpty: Boolean get() = lock.withLock { value === EMPTY }
+    protected final override val isBufferAlwaysFull: Boolean = false
```

Comment:
Both `isBufferAlwaysEmpty` and `isBufferAlwaysFull` should be written with `get() = false` as they were before to avoid an extra field that is not needed here.